src/mlpack/methods/xgboost/loss_functions/sse_loss.hpp

```diff
@@ -0,0 +1,126 @@
+/**
+ * @file methods/xgboost/loss_functions/sse_loss.hpp
+ * @author Rishabh Garg
+ *
+ * The sum of squared error loss class, which is a loss funtion for gradient
+ * xgboost based decision trees.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_XGBOOST_LOSS_FUNCTIONS_SSE_LOSS_HPP
+#define MLPACK_METHODS_XGBOOST_LOSS_FUNCTIONS_SSE_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ensemble {
+
+/**
+ * The SSE (Sum of Squared Errors) loss is a loss function to measure the
+ * quality of prediction of response values present in the node of each
+ * xgboost tree. It is also a good measure to compare the spread of two
+ * distributions. We will try to minimize this value while training.
+ *
+ * Loss = 1 / 2 * (Observed - Predicted)^2
+ */
+class SSELoss
+{
+ public:
+  /**
+   * Returns the initial predition for gradient boosting.
+   */
+  template<typename VecType>
+  typename VecType::elem_type InitialPrediction(const VecType& values)
+  {
+    return arma::accu(values) / (typename VecType::elem_type) values.n_elem;
+  }
```

Comment:
Can you clarify a bit about how or where this will be used?  I'm not sure I understand completely.

Reply:
Okay, let me go to the basics. In gradient boosting, we start with a constant prediction for all the data points. Then we calculate the residuals (difference between true and predicted values) and then we train multiple trees to fit those residuals.

So to begin with some residuals, we need some initial prediction and what value is given by this function. If you read any paper on gradient boosting, then this initial prediction is calculated by equating the derivative of loss function equal to zero.

I hope I answered your question. :)

Reply:
Thanks for the clarification. :+1: